### ISSUE TITLE
Bug: stuck on Welcome/Start Tour page on community.mattermost.com

### ISSUE THREAD

**Issue by Pinjasaur** [mattermost/focalboard#3995]
## Steps to reproduce the behavior

1. Try to go for Boards on community.mattermost.com
2. Stuck on welcome/start tour page (start or dismiss tour buttons do nothing)

## Expected behavior

I've previously dismissed the welcome tour so I shouldn't be seeing it. And if I was, I should be able to start or dismiss the tour.

## Screenshots (optional)

![image](https://user-images.githubusercontent.com/6335792/195118285-d7c35fdd-5d2e-4b70-b8f6-0ec669556970.png)

## Edition and Platform

 - Edition: Mattermost Boards (plugin)
 - Version: community (7.4.2)
 - Browser and OS: FF106 Mac

## Additional context (optional)

"Start tour" button doesn't appear to do anything based on lack of console or network tab activity. It may be breaking earlier on in the app lifecycle.

"Dismiss tour" button casues API call(s) that first return a 200 to the config, then a 2nd for the categories which has a malformed URL which directs to a 404. Captured below:

![image](https://user-images.githubusercontent.com/6335792/195118930-ff1d22ea-4834-44c2-859f-da0a5706f79a.png)

Malformed (double `//`):
![image](https://user-images.githubusercontent.com/6335792/195119084-b2e7161b-0f07-4fa5-9e36-8efc52fc4b0e.png)

Redirected to, which 404s:
![image](https://user-images.githubusercontent.com/6335792/195119184-964db8bd-eca0-4cdc-a2b9-d80884027e84.png)

**Comment by Pinjasaur** [comment]
Based on that malformed URL it looks like the Team ID is not defined? It's also not a Focalboard API. https://api.mattermost.com/#tag/channels/operation/GetSidebarCategoriesForTeamForUser

**Comment by Pinjasaur** [comment]
I'm wondering if the 404 is a bit of a [red herring](https://en.wikipedia.org/wiki/Red_herring). I'm at a loss to why it's broken. I don't see anything that stands out to me in terms of network errors (4xx or 5xx) or console errors. @ogi-m this is different from the behavior you've noticed, right?

**Comment by ogi-m** [comment]
@Pinjasaur Yeah it's different, in the case of https://github.com/mattermost/focalboard/issues/3993 I can open everything except for the product tour

**Comment by Pinjasaur** [comment]
I think it might be a localized problem to the state of my user somehow. Changing to private/incognito browser and trying a different browser (Safari) doesn't change the behavior: I'm still unable to use Boards.

**Comment by Pinjasaur** [comment]
Got to the root cause for me, I had a `hiddenBoardIDs` value of `"\"\""` which JSON.parses to a `""` literal which is not of type Array so `"".forEach` throws an error and the whole frontend breaks. I believe this is isolated to myself as back when hiding boards was a new feature I tried it out, realized I still wanted to be able to see the board, and manually patched my user props back to a `hiddenBoardIDs` of `""` literal. All was fine until the 7.4 migration to the Preferences table when that `""` got inserted literally so now the API was (correctly) returning value of `"\"\""` since the `""` needed to be encoded within JSON.

tl;dr user error, here's the patch payload that fixed it for me:

```
{"updatedFields":{"hiddenBoardIDs":"[]"}}
```

here's the `.forEach` line that was ultimately breaking: https://github.com/mattermost/focalboard/blob/edaa900cdeafee38da4f4b3d3fa4a744143457c7/webapp/src/user.tsx#L39

**Comment by Pinjasaur** [comment]
to be clear: to experience this you would need to manually `PUT` in a bad value for `hiddenBoardIDs`, just like I unfortunately did. it just took the migration to become apparent for me. even if you hide a board and unhide it, the `hiddenBoardIDs` value gets set back to a `[]`
